File: pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    long long card = get_long_long("Please input the Credit Card number: ");
    int last_digit = 0, second_last_digit = 0, num_digits = 0, sum_of_doubled_evens = 0, sum_of_odds = 0;

    while (card > 0)
    {
        second_last_digit = last_digit;
        last_digit = card % 10;

        if (num_digits % 2 == 0)
        {
            sum_of_odds += last_digit;
        }
        else
        {
            int multiple = 2 * last_digit;
            sum_of_doubled_evens += (multiple / 10) + (multiple % 10);
        }

        card /= 10;
        num_digits++;
    }

    bool is_valid = (sum_of_odds + sum_of_doubled_evens) % 10 == 0;
    int first_two_digits = (last_digit * 10) + second_last_digit;

    if (last_digit == 4 && num_digits >= 13 && num_digits <= 16 && is_valid)
    {
        printf("VISA\n");
    }
    else if (first_two_digits >= 51 && first_two_digits <= 55 && num_digits == 16 && is_valid)
    {
        printf("MASTERCARD\n");
    }
    else if ((first_two_digits == 34 || first_two_digits == 37) && num_digits == 15 && is_valid)
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: pset6/credit/credit.py
from cs50 import get_int

card = get_int('Please input the Credit Card number: ')
last_digit, second_last_digit, num_digits, sum_of_doubled_evens, sum_of_odds = 0, 0, 0, 0, 0

while card > 0:
    second_last_digit = last_digit
    last_digit = int(card % 10)

    if num_digits % 2 is 0:
        sum_of_odds += last_digit
    else:
        multiple = 2 * last_digit
        sum_of_doubled_evens += int((multiple / 10)) + int((multiple % 10))

    card = int(card/10)
    num_digits += 1

is_valid = (sum_of_odds + sum_of_doubled_evens) % 10 == 0
first_two_digits = (last_digit * 10) + second_last_digit

if last_digit == 4 and num_digits >= 13 and num_digits <= 16 and is_valid:
    print('VISA')

elif first_two_digits >= 51 and first_two_digits <= 55 and num_digits == 16 and is_valid:
    print('MASTERCARD')

elif (first_two_digits == 34 or first_two_digits == 37) and num_digits == 15 and is_valid:
    print('AMEX')

else:
    print('INVALID')